controllers/users.js

```diff
@@ -70,6 +70,36 @@ const getUsers = async (req, res) => {
     const query = req.query?.query ?? "";
     const qualifiers = getQualifiers(query);
 
+    // getting user details by id if present.
+    if (req.query.id) {
+      const id = req.query.id;
+      let result;
+      try {
+        result = await userQuery.fetchUser({ userId: id });
+      } catch (error) {
+        logger.error(`Error while fetching user: ${error}`);
+        return res.boom.serverUnavailable(SOMETHING_WENT_WRONG);
+      }
+
+      if (!result.userExists) {
+        return res.boom.notFound("User doesn't exist");
+      }
+
+      const user = { ...result.user };
+      try {
+        delete user.phone;
+        delete user.email;
+      } catch (error) {
+        logger.error(`Error while formatting phone and email: ${error}`);
+        return res.boom.badImplementation("Error while formatting phone and email");
+      }
```

Comment:
If the property doesn't exist, the delete operation simply does nothing and doesn't throw any error. we dont need this try catch block

Reply:
removed try catch block